app/models/add_tag_filter.rb

```diff
@@ -2,8 +2,12 @@
 
 class AddTagFilter < TagFilter
   def apply(item_ids = [])
-    items = item_ids.any? ? FeedItem.find(item_ids) :
-      scope.taggable_items
+    if item_ids.any?
+      item_ids &= scope.taggable_items.pluck(:id)
```

Comment:
Am I right in interpreting that this means the tag filter will be applied to all tags in the scope AND any passed in items? That means it's no longer possible to apply a filter to a subset of items within a scope. Unless I'm missing something, this also seems redundant because items that get passed in should be in the scope. This is to pass a test?

Reply:
> Am I right in interpreting that this means the tag filter will be applied to all tags in the scope AND any passed in items?

No, it will intersect them. It will be applied to all items in the scope from the passed items set only.

> That means it's no longer possible to apply a filter to a subset of items within a scope.

This should be still possible.

> Unless I'm missing something, this also seems redundant because items that get passed in should be in the scope. This is to pass a test?

It's a bug. Before that change it was possible to apply a filter to items out of the filter scope which is wrong and should never happen, I think so, correct me if I'm wrong. The same applies to all four filter types.

Here a related test: https://github.com/berkmancenter/tagteam/blob/dev/spec/support/shared_tag_filter_examples.rb#L236

Reply:
Gotcha. My bad. Yep, this look correct then.